### DISCUSSION TITLE
Performance benchmarks

### DISCUSSION THREAD

**Discussion by hariria** [Brendonovich/prisma-client-rust#220] [Q&A]
Hey I was curious if there were any performance benchmarks done on this vs. other competitors like Diesel / SeaORM etc. vs raw sql. Thanks!

**Comment by Brendonovich** [comment]
We've ran some benchmarks for compile time, but not really any for runtime. I can confidently say though that PCR will likely not be as fast as other ORMS, especially raw SQL. The Prisma engines do a loooot of work behind the scenes that other ORMs don't.

**Reply by hariria** [reply]
Makes sense. Would be really cool to see some runtime comparisons!

**Comment by hariria** [comment]
@Brendonovich I've made a very very simple benchmark. It could probably use some work and would love your input on it! https://github.com/hariria/rust-orm-bench. TLDR from running it:

```bash
# Each of the Create / Find / Delete benchmarks below 
# were based on 10,000 records being added, found, or deleted 
# to a SQLITE DB serially, without any kind of concurrency

# Each record was related to a user, with a GUID as its PK
# and ID. For more info see the prisma/schema.prisma file

Running diesel benchmarks...
Create users elapsed 4.08s
Find users elapsed 108.22ms
Delete users elapsed 4.21s


Running prisma benchmarks...
Create users elapsed 8.42s
Find users elapsed 2.37s
Delete users elapsed 7.63s


Running prisma raw sql benchmarks...
Create users elapsed 4.28s
Find users elapsed 1.35s
Delete users elapsed 4.20s
```

Most notably it seems like read performance for Diesel is much closer to the read performance limits of sqlite (~100k reads / second) and for some reason prisma was 20x slower. I tried using `tokio::spawn` to run some concurrent queries but that seemed to make no difference in performance
